<?xml version="1.0"?>
<launch>

    <arg name="robot" default="" />

    <arg name="model" default="four_diff"/>
    <arg name="turtle_model" default="waffle" />
    <arg name="x_pos" default="-2.0"/>
    <arg name="y_pos" default="-0.5"/>
    <arg name="z_pos" default="0.0"/>


    <include file="$(find turtlebot3_gazebo)/launch/turtlebot3_world.launch">
        <arg name="model" value="$(arg turtle_model)" />
        <arg name="x_pos" value="$(arg x_pos)" />
        <arg name="y_pos" value="$(arg y_pos)" />
        <arg name="z_pos" value="$(arg z_pos)" />
    </include>


    <node pkg="diff_control" type="diff_control_node" name="diff_controller" output="screen">
        <param name="vrpn_topic" value="vrpn_client_node/$(arg robot)/pose" />
        <param name="pose_topic" value="slam_toolbox/pose" />
        <param name="cmd_vel_topic" value="$(arg robot)/cmd_vel" />
        <param name="goal_topic" value="$(arg robot)/goal" />

        <!-- TurtleSim Parameters -->
        <param name="use_turtle_sim" value="false" />
        <param name="turtle_topic" value="turtle1/pose" />

        <!-- Controller Parameters -->
        <param name="a" value="0.1" />
        <param name="Kx" value="1.0" />
        <param name="Ky" value="1.0" />
        <param name="control_frequency" value="10.0" />
    </node>

</launch>